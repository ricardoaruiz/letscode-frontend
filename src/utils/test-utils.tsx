import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from '../styles/Theme'

const renderWithContext = (children: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'

export { renderWithContext }
