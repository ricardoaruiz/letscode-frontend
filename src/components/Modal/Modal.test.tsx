import React from 'react'
import { render, screen, fireEvent } from '../../utils/test-utils'

import { Modal } from '.'

describe('<Modal />', () => {
  it('should be render correctly', () => {
    const { container } = render(
      <Modal>
        <h3>Title</h3>
        <p>Content</p>
      </Modal>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be open state render successfuly', () => {
    const { rerender } = render(
      <Modal isOpen={true}>
        <h3>Title</h3>
        <p>Content</p>
      </Modal>
    )

    const modalContent = screen.getByRole('alertdialog')
    expect(modalContent).toHaveAttribute('aria-hidden', 'false')

    rerender(
      <Modal isOpen={false}>
        <h3>Title</h3>
        <p>Content</p>
      </Modal>
    )

    expect(modalContent).toHaveAttribute('aria-hidden', 'true')
  })

  it('should be open close modal when press ESC key', () => {
    const mockedCloseOnEsc = jest.fn()
    render(
      <Modal isOpen={true} closeOnEsc={mockedCloseOnEsc}>
        <h3>Title</h3>
        <p>Content</p>
      </Modal>
    )

    const modalContent = screen.getByRole('alertdialog')
    expect(modalContent).toHaveAttribute('aria-hidden', 'false')

    fireEvent.focus(modalContent)
    fireEvent.keyUp(modalContent, {
      key: 'Escape',
    })

    expect(mockedCloseOnEsc).toHaveBeenCalled()
  })
})
