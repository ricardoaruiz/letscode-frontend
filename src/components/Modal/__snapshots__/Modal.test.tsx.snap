// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should be render correctly 1`] = `
<div
  class="sc-bczRLJ cnUETY"
>
  <div
    aria-hidden="true"
    class="sc-gsnTZi bkcHLK"
    role="alertdialog"
    tabindex="1"
  >
    <h3>
      Title
    </h3>
    <p>
      Content
    </p>
  </div>
</div>
`
